homeassistant/helpers/entity_component.py

```diff
@@ -85,17 +86,22 @@ def _setup_platform(self, platform_type, platform_config,
             return
 
         # Config > Platform > Component
-        scan_interval = platform_config.get(
-            CONF_SCAN_INTERVAL,
-            getattr(platform, 'SCAN_INTERVAL', self.scan_interval))
+        scan_interval = (platform_config.get(CONF_SCAN_INTERVAL) or
+                         getattr(platform, 'SCAN_INTERVAL', None) or
+                         self.scan_interval)
         entity_namespace = platform_config.get(CONF_ENTITY_NAMESPACE)
 
+        key = (platform_type, scan_interval, entity_namespace)
```

Comment:
Do we want `scan_interval` in the key? This will mean there will be unused _platforms left over each time you change the scan interval

What about setting the new `scan_interval` in an `else:` block or after line 98? It is only used in `add_entities` and by this time `_unsub_polling` should be `None` and we can re-register the `track_time_change` handler